server/game/player.js

```diff
@@ -893,21 +894,11 @@ class Player extends Spectator {
     }
 
     moveCardToTopOfDeck(card, allowSave = true) {
-        this.game.applyGameAction('moveToTopOfDeck', card, card => {
-            this.game.raiseEvent('onCardReturnedToDeck', { player: this, card: card, allowSave: allowSave }, event => {
-                event.cardStateWhenMoved = card.createSnapshot();
-                this.moveCard(card, 'draw deck', { allowSave: allowSave });
-            });
-        });
+        return this.game.resolveGameAction(ReturnCardToDeck, { card, allowSave });
     }
 
     moveCardToBottomOfDeck(card, allowSave = true) {
-        this.game.applyGameAction('moveToBottomOfDeck', card, card => {
-            this.game.raiseEvent('onCardReturnedToDeck', { player: this, card: card, allowSave: allowSave }, event => {
-                event.cardStateWhenMoved = card.createSnapshot();
-                this.moveCard(card, 'draw deck', { bottom: true, allowSave: allowSave });
-            });
-        });
+        return this.game.resolveGameAction(ReturnCardToDeck, { card, allowSave, bottom: true });
```

Comment:
Might as well do `GameActions.returnCardToDeck` here like you did with shuffleIntoDeck, rather than include the specific game action.

Reply:
fixed